File: Desarrollo Software-DS/Aplicacion Android-Flutter/Código/filtroParteTrabajo.dart
import 'package:sosemado/tecnico.dart';

import 'filtro.dart';
import 'trabajo.dart';
import 'parteTrabajo.dart';



class FiltroParteTrabajo extends Filtro {
  parteTrabajo pt = new parteTrabajo(0.0, 0.0, 0.0, 0.0);

  FiltroParteTrabajo(Tecnico t) {
    pt=t.getParte();
  }

void filtrar(Trabajo t)
{
  
    if(pt.getDificultad()<5){
        t.setPresupuesto(t.getPresupuesto()-30);
    }
    else if(pt.getDificultad()>5 && pt.getDificultad()<8)
       t.setPresupuesto(t.getPresupuesto()+10);
    else
        t.setPresupuesto(t.getPresupuesto()+20);

    if(pt.getHoras()>4)
    t.setPresupuesto(t.getPresupuesto()+(5*pt.getHoras()));
    
    t.setPresupuesto(t.getPresupuesto()+pt.getGastosMateriales());
    t.setPresupuesto(t.getPresupuesto()+pt.getGastosTransporte());
    
}



}

File: Tercero/Desarrollo Software-DS/Practica 1/Sesion3/Obligatoria/S3/FiltroParteTrabajo.cpp
#include "FiltroParteTrabajo.h"

FiltroParteTrabajo::FiltroParteTrabajo()
{
    pt = new parteTrabajo(0,0,0,0);
    pt->rellenarParteTrabajo();
}

void FiltroParteTrabajo::filtrar(Trabajo * t)
{
  
    if(pt->getDificultad()<5){
        t->setPresupuesto(t->getPresupuesto()-30);
    }
    else if(pt->getDificultad()>5 && pt->getDificultad()<8)
       t->setPresupuesto(t->getPresupuesto()+10);
    else
        t->setPresupuesto(t->getPresupuesto()+20);

    if(pt->getHoras()>4)
    t->setPresupuesto(t->getPresupuesto()+(5*pt->getHoras()));
    
    t->setPresupuesto(t->getPresupuesto()+pt->getGastosMateriales());
    t->setPresupuesto(t->getPresupuesto()+pt->getGastosTransporte());
    
}